gsl/span

```diff
@@ -670,6 +670,19 @@ span<typename Ptr::element_type>
 make_span(Ptr& cont)
 { return span<typename Ptr::element_type>(cont); }
 
+template <class T, class = std::enable_if_t<std::is_pod<T>::value>>
```

Comment:
Actually I think we don't really need to restrict `as_bytes()` or `as_writeable_bytes()' at all (as per the `span` member functions). Isee these as the "sharp knife" power-user functions. ;-) People who are calling them should be aware of the language rules (and any compiler extensions they might have) around accessing object storage.